src/Testcontainers.Grafana/GrafanaBuilder.cs

```diff
@@ -0,0 +1,159 @@
+            .WithEnvironment("GF_SECURITY_ADMIN_USER", username);
+    }
+
+    /// <summary>
+    /// Sets the Grafana password.
+    /// </summary>
+    /// <param name="password">The Grafana password.</param>
+    /// <returns>A configured instance of <see cref="GrafanaBuilder" />.</returns>
+    public GrafanaBuilder WithPassword(string password)
+    {
+        return Merge(DockerResourceConfiguration, new GrafanaConfiguration(password: password))
+            .WithEnvironment("GF_SECURITY_ADMIN_PASSWORD", password);
+    }
+
+    /// <summary>
+    /// Disables the anonymous access.
+    /// </summary>
+    /// <returns>A configured instance of <see cref="GrafanaBuilder" />.</returns>
+    public GrafanaBuilder WithAnonymousAccessDisabled()
+    {
+        return WithEnvironment("GF_AUTH_ANONYMOUS_ENABLED", "false");
+    }
+
+    /// <summary>
+    /// Enables the anonymous access.
+    /// </summary>
+    /// <returns>A configured instance of <see cref="GrafanaBuilder" />.</returns>
+    public GrafanaBuilder WithAnonymousAccessEnabled()
+    {
+        return WithEnvironment("GF_AUTH_ANONYMOUS_ENABLED", "true")
+            .WithEnvironment("GF_AUTH_ANONYMOUS_ORG_ROLE", "Admin");
+    }
+
+    /// <summary>
+    /// Mounts a datasource configuration file.
+    /// </summary>
+    /// <param name="datasourceFilePath">The path to the datasource configuration file.</param>
+    /// <returns>A configured instance of <see cref="GrafanaBuilder" />.</returns>
+    public GrafanaBuilder WithDataSource(string datasourceFilePath)
+    {
+        return WithBindMount(datasourceFilePath, "/etc/grafana/provisioning/datasources/");
+    }
+
+    /// <summary>
+    /// Mounts a dashboard configuration file.
+    /// </summary>
+    /// <param name="dashboardFilePath">The path to the dashboard configuration file.</param>
+    /// <returns>A configured instance of <see cref="GrafanaBuilder" />.</returns>
+    public GrafanaBuilder WithDashboard(string dashboardFilePath)
+    {
+        return WithBindMount(dashboardFilePath, "/etc/grafana/provisioning/dashboards/");
+    }
+
+    /// <summary>
+    /// Mounts a plugin configuration file.
+    /// </summary>
+    /// <param name="pluginFilePath">The path to the plugin configuration file.</param>
+    /// <returns>A configured instance of <see cref="GrafanaBuilder" />.</returns>
+    public GrafanaBuilder WithPlugin(string pluginFilePath)
+    {
+        return WithBindMount(pluginFilePath, "/etc/grafana/provisioning/plugins/");
+    }
+
+    /// <summary>
+    /// Mounts a notifier configuration file.
+    /// </summary>
+    /// <param name="notifierFilePath">The path to the notifier configuration file.</param>
+    /// <returns>A configured instance of <see cref="GrafanaBuilder" />.</returns>
+    public GrafanaBuilder WithNotifier(string notifierFilePath)
+    {
+        return WithBindMount(notifierFilePath, "/etc/grafana/provisioning/notifiers/");
+    }
```

Comment:
These builder APIs haven't been tested. We typically add builder APIs only for configurations that are essential to run an instance or required for specific use cases.